nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/test/java/org/apache/nifi/web/api/OIDCAccessResourceTest.java

```diff
@@ -0,0 +1,155 @@
+        accessResource.oidcCallback(mockRequest, mockResponse);
+    }
+
+    public class MockOIDCCallbackFailure extends MockOIDCAccessResource {
+
+        public MockOIDCCallbackFailure(OidcService oidcService, Boolean requestShouldSucceed) throws IOException {
+            super(oidcService, requestShouldSucceed);
+        }
+
+        @Override
+        protected void forwardToLoginMessagePage(final HttpServletRequest httpServletRequest, final HttpServletResponse httpServletResponse, final String message) throws Exception {
+            assertEquals(OIDC_LOGIN_FAILURE_MESSAGE, message);
+        }
+    }
+
+    public class MockOIDCAccessResource extends OIDCAccessResource {
+
+        final static String BEARER_TOKEN = "bearer_token";
+        final static String AUTHORIZATION_CODE = "authorization_code";
+        final static String CALLBACK_URL = "https://nifi.apache.org/nifi-api/access/oidc/callback";
+        final static String RESOURCE_URI = "resource_uri";
+        private Boolean requestShouldSucceed;
+
+        public MockOIDCAccessResource(final OidcService oidcService, final Boolean requestShouldSucceed) throws IOException {
+            this.requestShouldSucceed = requestShouldSucceed;
+            final BearerTokenProvider bearerTokenProvider = Mockito.mock(StandardBearerTokenProvider.class);
+            Mockito.when(bearerTokenProvider.getBearerToken(any(LoginAuthenticationToken.class))).thenReturn(BEARER_TOKEN);
+            setOidcService(oidcService);
+            setBearerTokenProvider(bearerTokenProvider);
+            final LoginAuthenticationToken token = Mockito.mock(LoginAuthenticationToken.class);
+            Mockito.when(oidcService.exchangeAuthorizationCodeForLoginAuthenticationToken(any(AuthorizationGrant.class))).thenReturn(token);
+        }
+
+        @Override
+        protected AuthenticationResponse parseOidcResponse(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, boolean isLogin) {
+            if (requestShouldSucceed) {
+                return getSuccessResponse();
+            } else {
+                return getErrorResponse();
+            }
+        }
+
+        @Override
+        protected void checkOidcState(HttpServletResponse httpServletResponse, final String oidcRequestIdentifier, AuthenticationSuccessResponse successfulOidcResponse, boolean isLogin)
+                throws Exception {
+            // do nothing
+        }
+
+        @Override
+        protected String getOidcCallback() {
+            return CALLBACK_URL;
+        }
+
+        @Override
+        protected String generateResourceUri(final String... path) {
+            return RESOURCE_URI;
+        }
+
+        @Override
+        protected URI getCookieResourceUri() {
+            try {
+                return new URI(RESOURCE_URI);
+            } catch (URISyntaxException e) {
+                e.printStackTrace();
+                return null;
+            }
```

Comment:
It should be possible to change this to use `URI.create()` which does not throw a checked exception.
```suggestion
            URI.create(RESOURCE_URI);
```